tensorflow/python/distribute/input_lib.py

```diff
@@ -1465,11 +1454,21 @@ def __init__(self, dataset, worker, devices):
     self._worker = worker
     self._devices = devices
     self._element_spec = dataset.element_spec
+    self._options=options
     self._make_iterator()
 
   def _make_iterator(self):
     raise NotImplementedError("must be implemented in descendants")
 
+  def _format_data_list_with_options(self, data_list):
+    """change the data list to tuple type if required"""
+    if (self._options
+        and self._options.experimental_replication_mode == InputReplicationMode.PER_REPLICA
+            and not self._options.experimental_prefetch_to_device):
```

Comment:
nit: Can we add a comment here for the reader explaining why we need this? since the iterator code is removed from here it may be hard to understand right away.